TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/android/SaveMapDialog.java

```diff
@@ -42,34 +47,54 @@
     }
 
     @Override
-    public void onAttach(Activity activity) {
-        super.onAttach(activity);
-        mCallbackListener = (CallbackListener) activity;
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        mCallbackListener = (CallbackListener) context;
     }
 
     @Override
     public View onCreateView(LayoutInflater inflator, ViewGroup container,
                              Bundle savedInstanceState) {
         View dialogView = inflator.inflate(R.layout.dialog_save_map, null);
-        getDialog().setTitle("Enter the name of the map");
-        mNameEditText = (EditText) dialogView.findViewById(R.id.map_name);
-        dialogView.findViewById(R.id.save_map_ok).setOnClickListener(this);
+        getDialog().setTitle(R.string.save_map_dialog_title);
+        mOkButton = (Button) dialogView.findViewById(R.id.save_map_ok);
+        mOkButton.setOnClickListener(this);
         dialogView.findViewById(R.id.save_map_cancel).setOnClickListener(this);
-        setCancelable(false);
+        mNameEditText = (EditText) dialogView.findViewById(R.id.map_name);
+        mNameEditText.addTextChangedListener(new TextWatcher() {
```

Comment:
I think it would be helpful for a user to generate a default map name based on e.g. the current date and time: `map_20170704_142400`. Only if this default name is intentionally erased, the Ok button is disabled.